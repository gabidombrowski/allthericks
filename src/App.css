.App {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.Title {
  font-family: 'AdultSwim';
}

.Card {
  margin: 16px;
}

.CardContent {
  text-align: left;
}

.Stat {
  font-weight: 700;
}
