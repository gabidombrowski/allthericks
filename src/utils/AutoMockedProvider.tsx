import React, { ReactNode } from 'react'
//import { gql } from 'apollo-boost'
import { ApolloProvider } from 'react-apollo'
import { ApolloClient } from 'apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { SchemaLink } from 'apollo-link-schema'
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools'
import { printSchema, buildClientSchema } from 'graphql'
import introspectionResult from './schema.json'
import { CustomResolver } from "./mergeResolvers"

const AutoMockedProvider: React.FunctionComponent<{
  children: ReactNode
  mockResolvers?: CustomResolver
}> = ({ children, mockResolvers }) => {
  const schemaSDL = printSchema(
    buildClientSchema({ __schema: introspectionResult.__schema as any }),
  )

  const schema = makeExecutableSchema({
    typeDefs: schemaSDL,
    resolverValidationOptions: {
      requireResolversForResolveType: false,
    },
  })

  addMockFunctionsToSchema({
    schema,
    mocks: mockResolvers,
  })

  const client = new ApolloClient({
    link: new SchemaLink({ schema }),
    cache: new InMemoryCache(),
  })

  //  Useful for testing:

  //   const RICKS_QUERY = gql`
  //   {
  //     characters(page: 1, filter: { name: "rick" }) {
  //       info {
  //         count
  //       }
  //       results {
  //         image
  //         name
  //         species
  //         type
  //         origin {
  //           name
  //         }
  //       }
  //     }
  //   }
  // `

  //   client.query({ query: RICKS_QUERY }).then(result => {
  //     console.log('query: ', RICKS_QUERY)
  //     console.log('result: ', result)
  //   })

  return <ApolloProvider client={client}>{children}</ApolloProvider>
}

export default AutoMockedProvider