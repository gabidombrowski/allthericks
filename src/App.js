import React from 'react';
import { ApolloProvider } from '@apollo/react-hooks';
import './App.css';
import ApolloClient from 'apollo-boost';
import Cards from "./Cards"
import { Box } from '@material-ui/core';

const client = new ApolloClient({
  uri: 'https://rickandmortyapi.com/graphql/',
});

const App = () => (
  <ApolloProvider client={client}>
    <div className="App">
      <h1 className="Title">All the Ricks</h1>
      <Box display={"inline-flex"} flexWrap={"wrap"} justifyContent={"center"} m={3}>
        <Cards />
      </Box>
    </div>
  </ApolloProvider>
);

export default App;
