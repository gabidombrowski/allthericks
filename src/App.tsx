import React from 'react'
import { ApolloProvider } from '@apollo/react-hooks'
import './App.css'
import ApolloClient from 'apollo-boost'
import Cards from './Cards'
import Episodes from './Episodes'

const client = new ApolloClient({
  uri: 'https://rickandmortyapi.com/graphql/',
})

const App = () => (
  <ApolloProvider client={client}>
    <div className='App'>
      <h1 className='Title'>All the Ricks</h1>
      <Episodes />
      <Cards />
    </div>
  </ApolloProvider>
)

export default App;
