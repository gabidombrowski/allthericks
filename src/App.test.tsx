import React from "react";
import renderer from "react-test-renderer";
import wait from "waait";
import App from "./App";
import AutoMockedProvider from "utils/AutoMockedProvider";

it("renders", async () => {
  const tree = renderer.create(
    <AutoMockedProvider>
      <App />
    </AutoMockedProvider>
  );

  await wait(0);

  expect(tree).toMatchSnapshot();
});
