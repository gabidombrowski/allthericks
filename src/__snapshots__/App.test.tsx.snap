// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="App"
>
  <h1
    className="Title"
  >
    All the Ricks
  </h1>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Loading...
    </p>
  </div>
  <div
    className="MuiBox-root MuiBox-root-45"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Loading...
    </p>
  </div>
</div>
`;
